parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestLz4RawCodec.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.hadoop;
+
+import org.apache.parquet.hadoop.codec.*;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class TestLz4RawCodec {
+  @Test
+  public void TestLz4Raw() throws IOException {
+    // Reuse the snappy objects between test cases
+    Lz4RawCompressor compressor = new Lz4RawCompressor();
+    Lz4RawDecompressor decompressor = new Lz4RawDecompressor();
+
+    TestLz4Raw(compressor, decompressor, "");
+    TestLz4Raw(compressor, decompressor, "FooBar");
+    TestLz4Raw(compressor, decompressor, "FooBar1", "FooBar2");
+    TestLz4Raw(compressor, decompressor, "FooBar");
+    TestLz4Raw(compressor, decompressor, "a", "blahblahblah", "abcdef");
+    TestLz4Raw(compressor, decompressor, "");
+    TestLz4Raw(compressor, decompressor, "FooBar");
+  }
```

Comment:
Would be nice to also test with larger compressible payloads.

Reply:
Fixed